tests/test_plot.py

```diff
@@ -0,0 +1,415 @@
+    assert np.array_equal(ax.collections[1].get_facecolor(), props["color"])
+    assert ax.texts == []
+    validate_plot_general_properties(ax, props)
+
+
+def test_embedding_plot_text_integer_axis(embset):
+    emb = embset["red"]
+    fig, ax = mpl.pyplot.subplots()
+    emb.plot(kind="text", x_axis=1, y_axis=2)
+    props = {
+        "data": np.concatenate((emb.vector[1:2] + 0.01, emb.vector[2:3])),
+        "x_label": "Dimension 1",
+        "y_label": "Dimension 2",
+        "title": "",
+        "label": "red",
+        "aspect": "auto",
+        # Not applicable: type, color
+    }
+    assert np.array_equal(ax.texts[0].get_position(), props["data"])
+    assert ax.collections == []
+    assert ax.texts[0].get_text() == props["label"]
+    validate_plot_general_properties(ax, props)
+
+
+def test_embedding_plot_scatter_emb_axis(embset):
+    emb = embset["red"]
+    fig, ax = mpl.pyplot.subplots()
+    emb.plot(kind="scatter", x_axis=embset["blue"], y_axis=embset["green"])
+    props = {
+        "type": mpl.collections.PathCollection,
+        "data": np.array([emb > embset["blue"], emb > embset["green"]]),
+        "x_label": "blue",
+        "y_label": "green",
+        "color": mpl.colors.to_rgba_array("steelblue"),
+        "title": "",
+        "label": "red",
+        "aspect": "auto",
+    }
+    assert np.array_equal(ax.collections[0].get_offsets()[0], props["data"])
+    assert isinstance(ax.collections[0], props["type"])
+    assert ax.texts[0].get_text() == props["label"]
+    validate_plot_general_properties(ax, props)
+
+
+def test_embedding_plot_arrow_emb_axis(embset):
+    emb = embset["red"] + embset["yellow"]
+    fig, ax = mpl.pyplot.subplots()
+    emb.plot(
+        kind="arrow",
+        x_axis=embset["blue"],
+        y_axis=embset["green"],
+        color="yellow",
+        show_ops=True,
+        axis_option="equal",
+    )
+    props = {
+        "type": mpl.collections.PolyCollection,
+        "data": np.array([emb > embset["blue"], emb > embset["green"]]),
+        "x_label": "blue",
+        "y_label": "green",
+        "color": mpl.colors.to_rgba_array("yellow"),
+        "title": "",
+        "label": "(red + yellow)",
+        "aspect": 1.0,
+    }
+    UV = np.concatenate((ax.collections[1].U, ax.collections[1].V))
+    assert isinstance(ax.collections[1], props["type"])
+    assert np.array_equal(UV, props["data"])
+    assert np.array_equal(ax.collections[1].get_facecolor(), props["color"])
+    assert ax.texts[0].get_text() == props["label"]
+    validate_plot_general_properties(ax, props)
+
+
+def test_embedding_plot_raises_error_when_no_axis(embset):
+    emb = embset["red"]
+    with pytest.raises(ValueError, match="The `x_axis` value cannot be None"):
+        emb.plot()
+    with pytest.raises(ValueError, match="The `y_axis` value cannot be None"):
+        emb.plot(x_axis=0)
+
+
+## ===> EmbeddingSet.plot tests <============================================ ##
```

Comment:
Something is telling me it might be best to split up the tests into two separate files. The test file is pretty big as is.

Reply:
Honestly, I considered that but ended up not doing it hesitantly :smile: I think I didn't do it because they were all about the `plot` method and also the `validation_plot_general_properties` was commonly used. Do you think it's better to split it, then?

Reply:
In the other PR I've made a folder for testing visualisations. Probably best to separate the `matplotlib` based tests and the `altair` based tests. But given that we have a folder for `matplotlib` it feels preferable to add a small `common.py` where we might store useful functions that help us with tests. Scrolling all day can be a downer and if something goes wrong it can be productive if the tests files are all somewhat specific.